compiler/src/dotty/tools/MainGenericRunner.scala

```diff
@@ -149,6 +149,13 @@ object MainGenericRunner {
       process(remainingArgs, settings)
     case (o @ colorOption(_*)) :: tail =>
       process(tail, settings.withScalaArgs(o))
+    case "-e" :: expression :: tail =>
+      val tempScript = writeFile(s"@main def main(args: String *): Unit =\n  ${expression}")
```

Comment:
I would omit the newline character as a needless complication.

I don't remember whether scala 2 has to write a file, but I bet it doesn't. Is it necessary to write a file?

Tired recently of deleting scala temp files from my WSL /tmp due to running `scala` and dotty tests.

Reply:
oh, possibly newline and indent matters for braceless syntax.

Reply:
> I would omit the newline character as a needless complication.

Sounds good.

> Tired recently of deleting scala temp files from my WSL /tmp due to running `scala` and dotty tests.
Totally agree.  I reviewed the scala2 code, and it doesn't create a file, although it's has different Runners available.